### PR TITLE
12 - Parse PSUs and computer cooling

### PR DESCRIPTION
**Pull request by igoranuch**

### CODE REVIEW

File: functions/src/common/cleanComplexTable.ts [hunk 1]
```diff
@@ -1,11 +1,28 @@
...
+    .map((item: string, index: number) => {
+      item.includes('Socket')
+        ? () => {
+            let sockets = item;
+
+            for (let i = index + 1; i < cleanedSpecsTable.length; i++) {
+              if (cleanedSpecsTable[i]) {
+                sockets += `,${cleanedSpecsTable[i]}`;
+              } else break;
+            }
```
Comment (VHarastei) [inline functions/src/common/cleanComplexTable.ts]: use forEach

File: functions/src/common/cleanComplexTable.ts [hunk 2]
```diff
@@ -1,11 +1,28 @@
...
+  const normalizedSpecsTable = cleanedSpecsTable
+    .map((item: string, index: number) => {
+      item.includes('Socket')
+        ? () => {
+            let sockets = item;
+
+            for (let i = index + 1; i < cleanedSpecsTable.length; i++) {
+              if (cleanedSpecsTable[i]) {
+                sockets += `,${cleanedSpecsTable[i]}`;
+              } else break;
```
Comment (VHarastei) [inline functions/src/common/cleanComplexTable.ts]: remove else

File: functions/src/common/cleanComplexTable.ts [hunk 3]
```diff
@@ -1,11 +1,28 @@
...
+            }
+
+            return sockets;
+          }
+        : item;
+    })
+    .filter((item: string) => item.includes('\t'));
```
Comment (VHarastei) [inline functions/src/common/cleanComplexTable.ts]: why 2 includes? we can do this inside filter -> includes

File: functions/src/common/cleanComplexTable.ts (hunk 1 above)
Comment (ogavashi) [reply functions/src/common/cleanComplexTable.ts:17]: use forEach

Comment (hampachovsky) [reply functions/src/common/cleanComplexTable.ts:17]: use forEach

File: functions/src/psu/parsePSUpage.ts [hunk 4]
```diff
@@ -0,0 +1,94 @@
...
+      return;
+    }
+
+    const camelName = camelize(name);
+    specs.hasOwnProperty(camelName)
+      ? (specs[`${camelName}Dimensions`] = removeNonBreakingSpace(value))
+      : (specs[camelName] = removeNonBreakingSpace(value));
+  });
+
+  console.log(specs, name);
```
Comment (ogavashi) [inline functions/src/psu/parsePSUpage.ts:63]: remove console.log

Comment (ogavashi) [inline functions/src/psu/parsePSUpage.ts:63]:
```suggestion
```

Remove this console.log

**Review by ogavashi — CHANGES_REQUESTED** [review]

File: functions/src/psu/parsePSUpage.ts (hunk 4 above)
Comment (ogavashi) [inline functions/src/psu/parsePSUpage.ts]:
```suggestion
```
Remove console.log

**Review by ogavashi — CHANGES_REQUESTED** [review]

**Review by ogavashi — DISMISSED** [review]

File: functions/src/common/cleanComplexTable.ts [hunk 5]
```diff
@@ -1,11 +1,35 @@
...
+const cleanComplexTable = (rawSpecsTable: any) => {
+  const cleanedSpecsTable = rawSpecsTable.split('\n');
+
+  const normalizedSpecsTable = cleanedSpecsTable.filter((item: string) => {
+    if (item.includes('Socket')) {
+      const indexOfSocketSpec = cleanedSpecsTable.indexOf(item);
+      const lastIndexOfSocketSpec = cleanedSpecsTable.indexOf(
+        '',
+        indexOfSocketSpec,
+      );
```
Comment (VHarastei) [inline functions/src/common/cleanComplexTable.ts:13]: why you simply don`t do indexOfSocketSpec + 1?

Comment (igoranuch) [reply functions/src/common/cleanComplexTable.ts:13]: because i need index where sockets end, if i use indexOfSocketSpec + 1 there will be 2 sockets

**Review by ogavashi — APPROVED** [review]

**Review by VHarastei — APPROVED** [review]

**Pull request merged**